sdl_android/src/androidTest/java/com/smartdevicelink/test/api/audio/AudioStreamManagerTest.java

```diff
@@ -0,0 +1,313 @@
+                int sample_rate = test[key_sample_rate];
+                int sample_type = test[key_sample_type];
+                int sample_type_result = test[key_sample_type_result];
+
+                MediaFormat format = new MediaFormat();
+
+                format.setInteger(MediaFormat.KEY_CHANNEL_COUNT, channel_count);
+                format.setInteger(MediaFormat.KEY_SAMPLE_RATE, sample_rate);
+                format.setInteger(MediaFormat.KEY_PCM_ENCODING, sample_type);
+
+                onOutputFormatChangedMethod.invoke(mockDecoder, format);
+
+                assertEquals(channel_count, outputChannelCountField.getInt(mockDecoder));
+                assertEquals(sample_rate, outputSampleRateField.getInt(mockDecoder));
+                assertEquals(sample_type_result, outputSampleTypeField.getInt(mockDecoder));
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+            fail();
+        }
+    }
+
+    private Method getSampleAtTargetMethod() {
+        Method method = null;
+        try {
+            method = BaseAudioDecoder.class.getDeclaredMethod("sampleAtTargetTime",
+                    double.class, SampleBuffer.class, double.class, double.class, double.class);
+            method.setAccessible(true);
+        } catch (NoSuchMethodException e) {
+            e.printStackTrace();
+            fail();
+        }
+        return method;
+    }
+
+    private BaseAudioDecoder createMockBaseAudioDecoder() {
+        AudioDecoderListener listener = new AudioDecoderListener() {
+            @Override
+            public void onAudioDataAvailable(SampleBuffer sampleBuffer) {
+
+            }
+
+            @Override
+            public void onDecoderFinish() {
+
+            }
+
+            @Override
+            public void onDecoderError(Exception e) {
+
+            }
+        };
+
+        return new BaseAudioDecoder(getSampleFile("sample.mp3"), 16000, SampleType.SIGNED_16_BIT, listener) {
+            @Override
+            public void start() {
+
+            }
+        };
+    }
+
+    private File getSampleFile(String fileName) {
+        File file = new File(mContext.getCacheDir() + "/" + fileName);
+        try {
+            InputStream is = mContext.getAssets().open(fileName);
+            FileOutputStream fos = new FileOutputStream(file);
+            byte[] buffer = new byte[1 << 12];
+            int length;
+            do {
+                length = is.read(buffer);
+                if (length > 0) {
+                    fos.write(buffer, 0, length);
+                }
+            } while (length > 0);
+
+            fos.flush();
+            fos.close();
+            is.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+            fail();
+        }
+
+        return file;
+    }
+
+    public interface StreamManagerTestCallback {
+        void onTestResult(boolean isPassed, Exception e);
+    }
+
+    ISdl sdlInterface = new ISdl() {

```

Comment:
You can use mockito to mock this and add specific "answers" when certain methods are called. See here https://github.com/smartdevicelink/sdl_android/blob/feature/issue_782_Android_Managers/sdl_android/src/androidTest/java/com/smartdevicelink/api/FileManagerTests.java#L142-L144